desktop/ui/src/main/java/org/datacleaner/widgets/tree/SchemaTree.java

```diff
@@ -269,10 +273,22 @@ private DefaultMutableTreeNode createLibrary(final DefaultMutableTreeNode librar
         final Set<ComponentSuperCategory> superCategories = descriptorProvider.getComponentSuperCategories();
         for (ComponentSuperCategory superCategory : superCategories) {
             final DefaultMutableTreeNode schemaNode = new DefaultMutableTreeNode(superCategory);
-            libraryRoot.add(schemaNode);
             final Collection<? extends ComponentDescriptor<?>> componentDescriptors = descriptorProvider
                     .getComponentDescriptorsOfSuperCategory(superCategory);
 
+            final List<ComponentDescriptor<?>> filteredComponentDescriptors = new ArrayList<>();
+
+            for (ComponentDescriptor<?> componentDescriptor : componentDescriptors) {
+                final String displayName = componentDescriptor.getDisplayName();
+                if (displayName.toLowerCase().contains(_searchTerm.toLowerCase())) {
```

Comment:
I doubt as it is initialized in the field itself to an empty string. A better candidate would be displayName, but I added print outs to both these variables and they are not null when the panel is not visible. Also, not a single exception is printed to the console when the panel is gone, so no trace at all...